### DISCUSSION TITLE
Orange Pi 5 Plus: SPI and boot from eMMC

### DISCUSSION THREAD

**Discussion by sevlakoz** [Joshua-Riek/ubuntu-rockchip#341] [General]
I've a problem understanding Wiki for Orange Pi 5 Plus.
According to it "Booting directly from a USB or NVMe requires flashing U-Boot to the SPI."
eMMC isn't mentioned here, but without bootloader flashed to SPI system doesn't boot from eMMC.
Am I missing smth?

**Comment by Joshua-Riek** [comment]
The eMMC does not require the SPI to be flashed. It should boot just fine.

**Reply by sevlakoz** [reply]
Seems like I need to try one more time. Right now OS is on eMMC + U-boot is on SPI. It works just fine. Can just clean SPI for the purpose of testing?

**Reply by Joshua-Riek** [reply]
Yes, you can clean the spi with the below command.
```
sudo dd if=/dev/zero of=/dev/mtdblock0 count=4096 bs=512
```

**Reply by sevlakoz** [reply]
Yep, it works with empty SPI. But I really don't get why it wasn't before. Is it possible that there was some data in SPI that prevents booting?

**Reply by Joshua-Riek** [reply]
If there was some data in the SPI it could have prevented your system from booting.

**Reply by sevlakoz** [reply]
It was brand new board, so data in SPI were from manufacturer.

**Reply by cristianapas** [reply]
Hello,
I recently purchased an Orange Pi 5 Plus, and until I flashed the SPI, I was unable to get it to boot from the EMMC.

**Reply by Joshua-Riek** [reply]
Then there must be some pre flashed bootloader in the spi.

**Reply by sevlakoz** [reply]
Seems like cleaning SPI is official solution. Manual section 2.5 starts with "Note, after burning the image into eMMC, if the test finds that it cannot be started, please clear the SPIFlash and try again. For the method of clearing SPIFlash, please refer to the method of using RKDevTool to clear SPIFlash."

**Reply by RevBaker** [reply]
I too was confused about the u-boot / SPI interaction so had a conversation with Grok this morning. Here is a transcript that may be of some use.....
[u-boot and SPI.pdf](https://github.com/user-attachments/files/23778870/u-boot.and.SPI.pdf)
